lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1423,6 +1403,32 @@ def write_store(
         )
         return rval
 
+    @router.delete(
+        "/api/workflows/{workflow_id}",
+        summary="Add the deleted flag to a workflow.",
+    )
+    def delete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_delete = self.workflows_manager.get_stored_workflow(trans, workflow_id, check_ownership=True)
+        self.workflows_manager.delete(workflow_to_delete)
+        return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+    @router.post(
+        "/api/workflows/{workflow_id}/undelete",
+        summary="Remove the deleted flag from a workflow.",
+    )
+    def undelete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_undelete = self.workflows_manager.get_stored_workflow(trans, workflow_id, check_ownership=True)
```

Comment:
arg, sorry for the bad suggestion, `check_ownership` isn't implemented in the workflow manager